python/lancedb/table.py

```diff
@@ -673,6 +676,7 @@ def search(
             and also the "_distance" column which is the distance between the query
             vector and the returned vector.
         """
+        EVENTS("search")
```

Comment:
For a serious user, wouldn't we expect them to always hit the rate limit since search is a very frequent event? Does this make the analytics less useful?

Reply:
Yes a serious user is meant to hit the limits or we'll start making large network requests. We'd definitely have to treat these as the representation of usage and not the exact usage. 

It's a choice between - we don't track event-X OR out of the registered events, which is an approximation of actual usage, event-X grew by Y%.
Does that make sense?

Reply:
Or if we want to make things a little complex we can start setting event level rate limiting.. The tradeoff then would be that dashboarding will get complex as the scales will be different.

Reply:
i think it's fine for now. my main concern is just that search is likely the only event to run into this issue.
but let's keep it simple in this version and let's see what the actual data looks like first.

Reply:
SG!